### PR TITLE
Pluginify obsidian feeds!!

### PR DESCRIPTION
**Pull request by pashashocky**
SOO There's a lot here, at the moment it's not functional. 
This is how I'd propose to pluginify this thing and version the releases...

There are a lot of useful things here and moving to typescript as well as cleaning up the `view.js` which is becoming quite large will definitely clean this up.

### CODE REVIEW

File: src/view/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    if (!hasDataView) {
+      const error =
+        "Dataview plugin is not installed. Please install community plugin.";
+      renderError(this.containerEl, error);
+      throw new Error(error);
+    }
+    const dv = getAPI();
+
+    // TODO: view.js needs to be refactored...
+    dv.taskList(
```
Comment (pashashocky) [inline src/view/index.tsx:33]: This would be the entry point into `view.js` and the thing that would actually render the `dv.taskList`.

**Comment by pashashocky** [comment]
Just had a thought, that the quickest way to make it functional would be to wrap the `view.js` into a class an initialize it in `index.ts`...

**Comment by lukemt** [comment]
> Just had a thought, that the quickest way to make it functional would be to wrap the `view.js` into a class an initialize it in `index.ts`...

Yeah why not, keep it simple...

Another way of keeping it simple though would be indeed to refactor view.js into several files.

As a first step let's wrap it somehow so we have less migration work.

Looks pretty good from what I have seen (looked at the diff from my phone)!

**Comment by pashashocky** [comment]
There's a lot more messages waiting for you on discord when you have time. I'll work on this a little more in the evening, but this also needs a proper hook into re-rendering the dataview which at the moment it's not doing.

File: manifest.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "id": "obsidian-feeds",
+  "name": "Obsidian Feeds",
+  "version": "0.0.1",
+  "minAppVersion": "1.2.0",
+  "description": "Magic feeds dataview query for obsidian",
```
Comment (madx) [inline manifest.json]: [🔧](https://github.com/erikthedeveloper/code-review-emoji-guide#:~:text=:wrench:)
```suggestion
  "description": "Obsidian plugin to add magic feed views using Dataview",
```

File: manifest.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  "author": "Pash Shocky",
```
Comment (madx) [inline manifest.json]: [🔧](https://github.com/erikthedeveloper/code-review-emoji-guide#:~:text=:wrench:) We should probably put Lukas' name in here what do you think? 😁

File: package.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
+{
+  "name": "obsidian-note-gallery",
```
Comment (madx) [inline package.json]: [🔧](https://github.com/erikthedeveloper/code-review-emoji-guide#:~:text=:wrench:) 
image

File: package.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  "version": "0.0.7",
+  "description": "A masonry note gallery for Obsidian.",
```
Comment (madx) [inline package.json]: [🔧](https://github.com/erikthedeveloper/code-review-emoji-guide#:~:text=:wrench:) Needs to be updated too (actually the whole file)

File: src/view/index.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  ) {
+    super(containerEl);
+    this.settings = getSettings(src, app, containerEl, ctx);
+  }
+
+  async onload() {
+    const hasDataView = isPluginEnabled(this.app);
+    if (!hasDataView) {
+      const error =
+        "Dataview plugin is not installed. Please install community plugin.";
```
Comment (madx) [inline src/view/index.tsx]: [🔧](https://github.com/erikthedeveloper/code-review-emoji-guide#:~:text=:wrench:)
```suggestion
        "Dataview plugin is not installed. Please install it from Community plugins.";
```

**Comment by pashashocky** [comment]
@lukemt @madx this is ready for merge!

**Comment by lukemt** [comment]
I like it! Looks good! Will test a little as well.
btw, you can make a list of authors too ;)

**Comment by pashashocky** [comment]
> I like it! Looks good! Will test a little as well. btw, you can make a list of authors too ;)

How does one make a list of authors @lukemt? Oh do you mean the manifest?

**Comment by lukemt** [comment]
yea

> > I like it! Looks good! Will test a little as well. btw, you can make a list of authors too ;)
> 
> How does one make a list of authors @lukemt? Oh do you mean the manifest?

yeah the manifest, also if it works we could add it to the Community Plugins List

**Review by lukemt — APPROVED** [review]
LGTM

**Pull request merged**